### PR TITLE
Fix loading button popping

### PR DESCRIPTION
**Pull request by elsalindholm**
Home screen loading and play button have been combined into one button that changes its content/properties depending on whether the app is loading or not.

The loading button should:
- say 'loading'
- have the same size & style as the play button
- have no hover effects
- not do anything onClick
- not disappear once loading has finished

### CODE REVIEW

File: src/components/common/button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { observer } from 'mobx-react';
+import React from 'react';
+
+import './button.scss';
+
+interface ButtonProps {
```
Comment (MNewbigging) [inline src/components/common/button/Button.tsx:6]: some small changes to the props:

- rename onPlay to onClick
- exiting and loading should be optional
- add an optional className: string property - this gets added to the hardcoded 'button' classname in the comp

File: src/components/common/button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  exiting: boolean;
+  loading: boolean;
+}
+
+@observer
+export class Button extends React.PureComponent<ButtonProps> {
+  public render() {
+    const { onPlay, exiting, loading } = this.props;
+
+    const buttonText = loading ? 'loading' : 'play';
```
Comment (MNewbigging) [inline src/components/common/button/Button.tsx]: 'text' should be a prop for this button - so the button display the given text unless it's loading, in which case it always just says loading

File: src/components/common/button/button.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  animation-fill-mode: backwards;
+
+  &.loading {
+    &:hover {
+      cursor: default;
+      border-bottom: 6px solid $borderBotColour;
+    }
+  }
+
+  &.play {
```
Comment (MNewbigging) [inline src/components/common/button/button.scss]: can't call this play! rename to something more generic

File: src/components/web/home/HomeScreenWeb.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { observer } from 'mobx-react';
 import React from 'react';
 import { LetterStatus } from '../../../model/Letter';
```
Comment (MNewbigging) [inline src/components/web/home/HomeScreenWeb.tsx:4]: put a space below react import

File: src/components/common/button/button.scss [hunk 5]
```diff
@@ -0,0 +1,35 @@
+@import '../styles/palette.scss';
+
+.button {
+  width: 300px;
```
Comment (MNewbigging) [inline src/components/common/button/button.scss]: button sizes cannot be in the generic styles, they must be in the component-style file that creates the button!

**Review by MNewbigging — CHANGES_REQUESTED** [review]

**Comment by elsalindholm** [comment]
Requested changes made, hopefully correctly!

File: src/components/common/button/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+@observer
+export class Button extends React.PureComponent<ButtonProps> {
+  public render() {
+    const { onClick, className, exiting, loading, buttonText } = this.props;
+
+    const btnText = loading ? 'loading' : buttonText;
+    const exitingClass = exiting ? 'exiting' : '';
+    const buttonClass = loading ? 'loading' : className;
```
Comment (MNewbigging) [inline src/components/common/button/Button.tsx]: you'd want to apply the given class name at all times, regardless of loading - custom class names should control things like width/height, which shouldn't change with loading

File: src/components/web/home/home-screen-web.scss [hunk 7]
```diff
@@ -49,34 +50,37 @@
     }
   }
 
-  .play {
+  .button {
```
Comment (MNewbigging) [inline src/components/web/home/home-screen-web.scss]: revert this to 'play'! this is your custom classname for the button, so should style against that

File: src/components/web/home/home-screen-web.scss [hunk 8]
```diff
@@ -49,34 +50,37 @@
...
     animation: slideInUp 0.4s cubic-bezier(0.175, 0.885, 0.32, 1.05) 0.8s;
     animation-fill-mode: backwards;
 
-    &:hover {
-      margin-top: 0px;
-      width: 310px;
-      height: 65px;
-      font-size: 2.1rem;
-      cursor: pointer;
+    &.loading {
```
Comment (MNewbigging) [inline src/components/web/home/home-screen-web.scss]: by doing this outside the button.scss, it means you'll have to do it again for every button that loads. What to do is:

- in Button.tsx, apply a className 'loading' if you're loading (you're kind of doing this, but shouldn't prevent their custom classname from being added, so make it a separate line)
- then in button.scss, you can say the above; if loading, don't do hover styles

File: src/components/web/home/home-screen-web.scss [hunk 9]
```diff
@@ -49,34 +50,37 @@
...
+        border-bottom: 6px solid $borderBotColour;
+      }
     }
 
-    &.exiting {
-      animation: slideOutDown 0.8s cubic-bezier(0.68, -0.55, 0.265, 1.55) 0.2s;
-      animation-fill-mode: forwards;
-    }
-  }
+    &.play {
```
Comment (MNewbigging) [inline src/components/web/home/home-screen-web.scss]: once you've renamed the above classname to 'play' from 'button' you can lose the top &.play here and just have the hover stuff inside

File: src/components/web/home/home-screen-web.scss [hunk 10]
```diff
@@ -49,34 +50,37 @@
...
+        font-size: 2.1rem;
+        cursor: pointer;
+      }
 
-  .loading {
-    width: 300px;
-    height: 60px;
-    margin-top: 5px;
-    font-size: 2rem;
+      &.exiting {
```
Comment (MNewbigging) [inline src/components/web/home/home-screen-web.scss]: put spaces above nested blocks like this!

**Comment by elsalindholm** [comment]
Please could you merge thsi request when you have a moment? :)

**Review by MNewbigging — APPROVED** [review]

**Pull request merged**